cpp/arcticdb/storage/mongo/mongo_client.cpp

```diff
@@ -154,6 +156,31 @@ auto build_document(storage::KeySegmentPair &kv) {
 
     return basic_builder.extract();
 }
+
+/*
+ * Mongo error handling notes:
+ * All exceptions are subtypes of https://mongocxx.org/api/current/classmongocxx_1_1exception.html
+ * - mongocxx::logic_error is thrown for invalid API calls and has no error number
+ * - mongocxx::operation_exception for wrapping errors from libmongoc with an std::error_code
+ *   See https://mongoc.org/libmongoc/current/errors.html for docs
+ *   https://github.com/mongodb/mongo-cxx-driver/blob/master/src/mongocxx/exception/error_code.hpp for constants
+ */
+
+/** Mongo often fail to set the what() string, so has to manually format the exception */
+[[noreturn]] void translate_operation_exception(const mongocxx::operation_exception& e) {
+    std::string json;
```

Comment:
(Minor) Isn't the naming slightly misleading? As far as I understand `bsoncxx::to_json(e.raw_server_error()->view());` would return a stringified JSON, but `fmt::format("(Failed to format server response either: {})", e.what());` `"No response from server";` are not JSONs.